chain/provider/provider.go

```diff
@@ -0,0 +1,67 @@
+package provider
+
+import (
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// BandProvider contain context, txBuilder, private key, and address
+type BandProvider struct {
+	cliCtx  context.CLIContext
+	txBldr  authtypes.TxBuilder
+	addr    sdk.AccAddress
+	privKey crypto.PrivKey
+}
+
+const Bech32MainPrefix = "band"
+
+func SetBech32AddressPrefixes(config *sdk.Config) {
+	config.SetBech32PrefixForAccount(Bech32MainPrefix, Bech32MainPrefix+sdk.PrefixPublic)
+	config.SetBech32PrefixForValidator(Bech32MainPrefix+sdk.PrefixValidator+sdk.PrefixOperator, Bech32MainPrefix+sdk.PrefixValidator+sdk.PrefixOperator+sdk.PrefixPublic)
+	config.SetBech32PrefixForConsensusNode(Bech32MainPrefix+sdk.PrefixValidator+sdk.PrefixConsensus, Bech32MainPrefix+sdk.PrefixValidator+sdk.PrefixConsensus+sdk.PrefixPublic)
+}
+
+func privKeyToBandAccAddress(privKey crypto.PrivKey) sdk.AccAddress {
+	config := sdk.GetConfig()
+	SetBech32AddressPrefixes(config)
+	return sdk.AccAddress(privKey.PubKey().Address().Bytes())
+}
+
+// NewBandProvider create new BandProvider create new cliCtx and txBldr
+func NewBandProvider(privKey crypto.PrivKey) BandProvider {
+	cdc := NewCodec()
+	addr := privKeyToBandAccAddress(privKey)
+	cliCtx := NewCLIContext(addr).WithCodec(cdc)
+	num, _, err := authtypes.NewAccountRetriever(cliCtx).GetAccountNumberSequence(addr)
+	if err != nil {
+		panic("Cannot get account number")
+	}
+
+	return BandProvider{
+		cliCtx:  cliCtx,
+		txBldr:  NewTxBuilder(utils.GetTxEncoder(cdc)).WithAccountNumber(num),
+		addr:    addr,
+		privKey: privKey,
+	}
+}
+
+func (provider *BandProvider) Sender() sdk.AccAddress {
+	return provider.addr
+}
+
+func (provider *BandProvider) SendTransaction(
+	msgs []sdk.Msg, seq, gas uint64,
+	memo, fees, gasPrices, broadcastMode string,
+) (sdk.TxResponse, error) {
+	cliCtx := provider.cliCtx.WithBroadcastMode(broadcastMode)
```

Comment:
What happens if someone passes a bogus `broadcastMode?`

Reply:
It will return error when broadcast transaction code in this [file](https://github.com/cosmos/cosmos-sdk/blob/master/client/context/broadcast.go#L31)